kernel/src/services.rs

```diff
@@ -1060,59 +1076,63 @@ impl SystemServices {
 
         // Save state if the PID has changed.  This will activate the new memory
         // space.
-        let new = self.get_process_mut(new_pid)?;
         if new_pid != previous_pid {
-            klog!("New process original state: {:?}", new.state);
+            {
```

Comment:
Why the extra block? Is it to prevent `new` from shadowing the one that appears below?